lib/helpers/model/discriminator.js

```diff
@@ -153,31 +153,36 @@ module.exports = function discriminator(model, name, schema, tiedValue, applyPlu
     if (baseSchema.options.collection) {
       schema.options.collection = baseSchema.options.collection;
     }
-
     const toJSON = schema.options.toJSON;
     const toObject = schema.options.toObject;
     const _id = schema.options._id;
     const id = schema.options.id;
 
     const keys = Object.keys(schema.options);
     schema.options.discriminatorKey = baseSchema.options.discriminatorKey;
-
+    const hasUserProvidedTypeKey = schema._userProvidedOptions.typeKey != null && baseSchema._userProvidedOptions.typeKey != null;
```

Comment:
This is a step in the right direction, but I'd like for you to generalize this check for `typeKey` to all schema options. So if user didn't specify other options in discriminator schema (capped, etc.) then use the base schema's options by default.

Reply:
Making the changes you requested causes the test mentioned earlier in the thread from failing.